cmd/webhook/main.go

```diff
@@ -102,31 +106,33 @@ func main() {
-		Client:  kubeClient,
-		Options: options,
-		Handlers: map[schema.GroupVersionKind]webhook.GenericCRD{
-			v1alpha1.SchemeGroupVersion.WithKind("Revision"):                 &v1alpha1.Revision{},
-			v1alpha1.SchemeGroupVersion.WithKind("Configuration"):            &v1alpha1.Configuration{},
-			v1alpha1.SchemeGroupVersion.WithKind("Route"):                    &v1alpha1.Route{},
-			v1alpha1.SchemeGroupVersion.WithKind("Service"):                  &v1alpha1.Service{},
-			v1beta1.SchemeGroupVersion.WithKind("Revision"):                  &v1beta1.Revision{},
-			v1beta1.SchemeGroupVersion.WithKind("Configuration"):             &v1beta1.Configuration{},
-			v1beta1.SchemeGroupVersion.WithKind("Route"):                     &v1beta1.Route{},
-			v1beta1.SchemeGroupVersion.WithKind("Service"):                   &v1beta1.Service{},
-			autoscalingv1alpha1.SchemeGroupVersion.WithKind("PodAutoscaler"): &autoscalingv1alpha1.PodAutoscaler{},
-			net.SchemeGroupVersion.WithKind("Certificate"):                   &net.Certificate{},
-			net.SchemeGroupVersion.WithKind("ClusterIngress"):                &net.ClusterIngress{},
-			net.SchemeGroupVersion.WithKind("ServerlessService"):             &net.ServerlessService{},
-		},
-		Logger:                logger,
-		DisallowUnknownFields: true,
-
-		// Decorate contexts with the current state of the config.
-		WithContext: func(ctx context.Context) context.Context {
-			return v1beta1.WithUpgradeViaDefaulting(store.ToContext(ctx))
-		},
+
+	handlers := map[schema.GroupVersionKind]webhook.GenericCRD{
+		v1alpha1.SchemeGroupVersion.WithKind("Revision"):                 &v1alpha1.Revision{},
+		v1alpha1.SchemeGroupVersion.WithKind("Configuration"):            &v1alpha1.Configuration{},
+		v1alpha1.SchemeGroupVersion.WithKind("Route"):                    &v1alpha1.Route{},
+		v1alpha1.SchemeGroupVersion.WithKind("Service"):                  &v1alpha1.Service{},
+		v1beta1.SchemeGroupVersion.WithKind("Revision"):                  &v1beta1.Revision{},
+		v1beta1.SchemeGroupVersion.WithKind("Configuration"):             &v1beta1.Configuration{},
+		v1beta1.SchemeGroupVersion.WithKind("Route"):                     &v1beta1.Route{},
+		v1beta1.SchemeGroupVersion.WithKind("Service"):                   &v1beta1.Service{},
+		autoscalingv1alpha1.SchemeGroupVersion.WithKind("PodAutoscaler"): &autoscalingv1alpha1.PodAutoscaler{},
+		net.SchemeGroupVersion.WithKind("Certificate"):                   &net.Certificate{},
+		net.SchemeGroupVersion.WithKind("ClusterIngress"):                &net.ClusterIngress{},
+		net.SchemeGroupVersion.WithKind("ServerlessService"):             &net.ServerlessService{},
+	}
+
+	// Decorate contexts with the current state of the config.
+	ctxFunc := func(ctx context.Context) context.Context {
+		return v1beta1.WithUpgradeViaDefaulting(store.ToContext(ctx))
+	}
+
+	controller, err := webhook.NewAdmissionController(kubeClient, options, handlers, logger, ctxFunc, true)
```

Comment:
It's unfortunate that "DisallowUnknownFields" gets collapsed to `true` in this constructor. Not for this PR, but it would be nice to have a signature like:

```golang
controller, err := webhook.AdmissionController(
	kubeClient,
	handlers,
	logger,
	webhook.EnrichContext(ctxFunc),
	webhook.Namespace(system.Namespace()),
	webhook.Secret("webhook-certs"),
	webhook.WebhookName("webhook.serving.knative.dev"),
	webhook.UnknownFields)
```

Where the default context function, port, and some other items were already set, and we only needed to override some fields. Since you're now returning an `error`, you could also complain at this point if Secret, Name, etc were not set.